src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,41 +1,64 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { closeСurrentTodo } from '../../features/currentTodo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
 
 export const TodoModal: React.FC = () => {
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
+
+  const [isLoading, setIsLoading] = useState(true);
+  const [user, setUser] = useState<User>();
+
+  useEffect(() => {
+    setIsLoading(true);
+    if (currentTodo) {
+      getUser(currentTodo.userId)
+        .then(setUser)
+        .finally(() => setIsLoading(false));
+    }
+  }, [currentTodo]);
```

Comment:
You're using a `useEffect` hook to fetch the user data whenever the `currentTodo` changes. However, you're not considering the case where `currentTodo` might be `undefined`. You should add a check for this case and handle it properly.